search/includes/classes/class-queue.php

```diff
@@ -499,4 +508,54 @@ public function action__ep_after_bulk_index( $document_ids, $slug, $return ) {
 
 		return true;
 	}
+
+	/**
+	 * Hook pre_ep_index_sync_queue for indexing operation ratelimiting.
+	 * If ratelimited, bail and offload indexing to queue.
+	 *
+	 * @param {bool} $bail Current value of bail
+	 * @param {object} $sync_manager Instance of Sync_Manager
+	 * @param {string} $indexable_slug Indexable slug
+	 * @return {bool} Whether to bail on indexing or not
+	 */
+	public function ratelimit_indexing( $bail, $sync_manager, $indexable_slug ) {
+		// Only posts supported right now
+		if ( 'post' !== $indexable_slug ) {
+			return $bail;
+		}
+
+		if ( empty( $sync_manager->sync_queue ) ) {
+			return $bail;
+		}
+
+		// Determine whether indexing is ratelimited currently and if this is an indexing operation
+		$is_indexing_ratelimited = false !== wp_cache_get( self::INDEX_QUEUEING_ENABLED_KEY, self::INDEX_COUNT_CACHE_GROUP );
+		
+		if ( $this->is_enabled() ) {
```

Comment:
Do we actually need this extra check? The filter won't be added if the queue is not enabled.